infra/batch/datasets/main.py

```diff
@@ -0,0 +1,225 @@
+import base64
+import gc
+import json
+import os
+import traceback
+import uuid
+import functions_framework
+import urllib3
+from sqlalchemy import create_engine, text
+from requests.exceptions import HTTPError
+
+import requests
+from google.cloud import storage
+from datetime import datetime
+from hashlib import md5
+from cloudevents.http import CloudEvent
+from google.cloud import pubsub_v1
+
+
+def upload_dataset(url, bucket_name, stable_id):
+    """
+    Uploads a dataset to a GCP bucket
+    :param url: dataset feed's producer url
+    :param bucket_name: name of the GCP bucket
+    :param stable_id: the dataset stable id
+    :return: the file hash and the hosted url as a tuple
+    """
+    # Fix DH Key issues in server side
+    requests.packages.urllib3.disable_warnings()
+    requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS += ':HIGH:!DH:!aNULL'
+    try:
+        requests.packages.urllib3.contrib.pyopenssl.util.ssl_.DEFAULT_CIPHERS += ':HIGH:!DH:!aNULL'
+    except AttributeError:
+        # no pyopenssl support used / needed / available
+        pass
+
+    headers = {
+        'User-Agent':
+            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) '
+            'Chrome/39.0.2171.95 Safari/537.36'
+    }
+    response = requests.get(url, headers=headers, verify=False)
+    response.raise_for_status()
+
+    content = response.content
```

Comment:
Could the size of the file cause problems since we load it all in memory?

Reply:
I tried to load data in chunks and compute the hash with a combination of its chunks which worked but didn't improve memory usage much.

Reply:
For the feed we currently have with the configuration I set up we do not currently have any memory problem. We should probably keep an eye on that!